### ISSUE TITLE
a Bug that hapens after few min

### ISSUE THREAD

**Issue by pigasspig** [leftwm/leftwm#456]
**Describe the bug**
This bug happens in 10-20 minutes after loading leftwm up so basically it makes me not able to type on anything like discord i cant type also cant use any binds and cant type on terminal but i can interact like leave a call mute go to a different server

**To Reproduce**
1. there are no steps it happens randomly

**Expected behavior**
i didnt expect anything

**Screenshots**

**Environment:**
 - OS/Distro: arch 
 - Output of `leftwm-check` i cant get because cant type in term .-.

**Additional context**

**Comment by VuiMuich** [comment]
Can you still access tty?
You should be able to run `leftwm check` from there. 

Am I getting it right, tja basically keyboard input gets stuck? Could this error to anything else like a compositor etc.?

**Comment by pigasspig** [comment]
i can

**Comment by pigasspig** [comment]
> Can you still access tty?
> You should be able to run `leftwm check` from there.
> 
> Am I getting it right, tja basically keyboard input gets stuck? Could this error to anything else like a compositor etc.?

i can also i cant type anyhting on anything when im on leftwm but it happens after a while

**Comment by VuiMuich** [comment]
Is leftwm responding to commands like `leftwm command NextLayout` etc. (list all possible commands with `-l` flag) from tty?
Can you open new windows from tty with e.g. `DISPLAY=:0 alacritty`?
Can you find any other pattern that might cause unresponsiveness by keyboard input? (If I got you right earlier, mouse input is still working, correct?)

**Comment by pigasspig** [comment]
> Is leftwm responding to commands like `leftwm command NextLayout` etc. (list all possible commands with `-l` flag) from tty?
> Can you open new windows from tty with e.g. `DISPLAY=:0 alacritty`?
> Can you find any other pattern that might cause unresponsiveness by keyboard input? (If I got you right earlier, mouse input is still working, correct?)

ye mouse input works i cant find any other pattern
 its just random

**Comment by VuiMuich** [comment]
> > Can you still access tty?
> 
> > You should be able to run `leftwm check` from there.
> 
> > 
> 
> > Am I getting it right, tja basically keyboard input gets stuck? Could this error to anything else like a compositor etc.?
> 
> 
> 
> i can also i cant type anyhting on anything when im on leftwm but it happens after a while
> 
> 

Then I'm a bit confused what the actual problem is?
Leftwm keybinds stop working?

**Comment by pigasspig** [comment]
its that and  i cant type anything on discord  or terminal 
and im waiting for the bug to happen so i can to leftwm check

**Comment by VuiMuich** [comment]
> its that and  i cant type anything on discord  or terminal 
> 
> and im waiting for the bug to happen so i can to leftwm check 

I don't think it's actually helpful to do leftwm check while your system has the issue, as it simply does a static check on your config.toml

**Comment by pigasspig** [comment]
ok

**Comment by pigasspig** [comment]
is there a way to fix the issue tho

**Comment by VuiMuich** [comment]
Nevertheless it might be helpful to wait until the 'keyboard freeze' occurs, to test if leftwm is actually frozen by trying `leftwm command` from tty.
I am sorry but I can't shake the feeling that something else and on a first glance completely unrelated is causing the issue. Such random phenomena have a tendency to be caused by little errors.

**Comment by pigasspig** [comment]
> Nevertheless it might be helpful to wait until the 'keyboard freeze' occurs, to test if leftwm is actually frozen by trying `leftwm command` from tty.
> I am sorry but I can't shake the feeling that something else and on a first glance completely unrelated is causing the issue. Such random phenomena have a tendency to be caused by little errors.

its been around 25-30 minutes and it hasnt happened yet

**Comment by mgabs** [comment]
Can you write a script to run the command "leftwm check" and sleep every couple of minutes ?
Write to a file and go back to using the PC normally.

**Comment by VuiMuich** [comment]
> Can you write a script to run the command "leftwm check" and sleep every couple of minutes ?
> Write to a file and go back to using the PC normally.

As I said, leftwm-check just does some static checks on config files and the environment. There is not much point in logging regularly or during the 'bug situation'.

**Comment by TotalDarkness-NRF** [comment]
I have been having the same issue when I run discord. If I switch focus from discord after a few mins discord either closes or freezes if I try to interact with it again. Sometimes this causes all input from keyboard to be unresponsive and I sometimes get stuck in some windows.

Usually after I can force the discord window close using the discord system tray or my mouse to open programs using polybar and rofi I can force close it easily and then stuff begins to work again. I have only had one time where I could not do anything and that was when I had a window in fullscreen.

I believe it an issue with running discord but can not confirm. This issue usually happens if I leave discord out of focus either by focusing on another window or switching to another tag that discord is not on (usually happens when switching tags)

**Comment by VuiMuich** [comment]
@TotalDarkness-NRF thanks, this looks like a useful clue to me.

When discord froze on you in fullscreen, how did you solve it? Were you able to access tty and kill discord from there?

Maybe this is related: https://www.reddit.com/r/ManjaroLinux/comments/deo4x2/discord_freezes_when_getting_notifications/
@pigasspig do you have any notification daemon installed?

**Comment by TotalDarkness-NRF** [comment]
> When discord froze on you in fullscreen, how did you solve it? Were you able to access tty and kill discord from there?
I had no way of accessing tty as the keyboard was unresponsive. I ended up just restarting my computer after a few minutes of trying.

> Maybe this is related: https://www.reddit.com/r/ManjaroLinux/comments/deo4x2/discord_freezes_when_getting_notifications/
This may be related. I did not have any notification daemon running. I will try dunst with libnotify as see if that fixes any issues.

**Comment by hertg** [comment]
@TotalDarkness-NRF 
Did installing and running dunst resolve your issue?
Because I had a similar issue with [Ferdi](https://getferdi.com/) and didn't realize that I accidentally removed dunst from my autostart when switching to LeftWM. Autostarting dunst again actually resolved the issue for me. :)

**Comment by TotalDarkness-NRF** [comment]
Yes installing and running dunst did resolve my issue. I believe I also installed libnotify as well as that is what discord could not find

**Comment by VuiMuich** [comment]
@pigasspig is this issue resolved for you now as well?
